src/execute.rs

```diff
@@ -44,6 +53,26 @@ pub fn try_liquid_staking(
         delegator_address: env.contract.address.to_string(),
     };
 
+    // query exchange rate
+    let q = QueryExchangeRateRequest {
+        chain_id: config.chain_id,
+    };
+    let exchange_rate_response: QueryExchangeRateResponse =
+        deps.querier.query(&QueryRequest::Stargate {
+            path: LIQUIDSTAKEIBC_RATE_QUERY_TYPE.to_string(),
+            data: q.into(),
+        })?;
+    let exchange_rate = exchange_rate_response.rate;
+
+    // calculate staked amount to be sent to the receiver
+    let exchange_rate_decimal = Decimal::from_str(&exchange_rate)?;
+    let amount_decimal = Decimal::from_str(&amount.to_string())?;
+    let staked_amount_decimal = amount_decimal.checked_mul(exchange_rate_decimal)?;
+
+    // convert decimal to Uint128 to be sent to the receiver and
+    let staked_amount = Decimal::to_uint_floor(staked_amount_decimal);
```

Comment:
let's call it `lst_mint_amount` instead?

Reply:
Done.